app/models/shipit/stack.rb

```diff
@@ -285,7 +285,7 @@ def last_completed_deploy
     end
 
     def previous_successful_deploy(deploy_id)
-      deploys_and_rollbacks.previous_successful(deploy_id)
+      deploys_and_rollbacks.success.where("id < ?", deploy_id).last
```

Comment:
this query will have the `stack` scope now?

Reply:
I believe so. All of my testing seems to show it as such, since is executing within the context of the relation to this stack.
This is the query generated (both by page load and by manually calling this method):
`Shipit::Task Load (0.3ms)`
```sql
 SELECT  "tasks".* FROM "tasks" WHERE "tasks"."stack_id" = ? AND "tasks"."type" IN (?, ?) AND "tasks"."status" = ? AND (id < 199) ORDER BY "tasks"."id" DESC LIMIT ?  [["stack_id", 4], ["type", "Shipit::Deploy"], ["type", "Shipit::Rollback"], ["status", "success"], ["LIMIT", 1]]
```

Reply:
👍